src/GridFS/CollectionWrapper.php

```diff
@@ -334,6 +342,32 @@ private function ensureIndexes()
         $this->ensureChunksIndex();
     }
 
+    private function indexKeysMatch(array $expectedKeys, array $actualKeys) : bool
+    {
+        if (count($expectedKeys) !== count($actualKeys)) {
+            return false;
+        }
+
+        $iterator = new MultipleIterator(MultipleIterator::MIT_NEED_ANY);
+        $iterator->attachIterator(new ArrayIterator($expectedKeys));
+        $iterator->attachIterator(new ArrayIterator($actualKeys));
+
+        foreach ($iterator as $key => $value) {
+            list($expectedKey, $actualKey)     = $key;
+            list($expectedValue, $actualValue) = $value;
+
+            if ($expectedKey !== $actualKey) {
+                return false;
+            }
+
+            if (! is_numeric($actualValue) || (int) $expectedValue !== (int) $actualValue) {
```

Comment:
If the actual value is not numeric, this function will return false even if values are identical, right? Maybe worth noting that somewhere.

Reply:
Added a clarifying comment. Thanks!